### PR TITLE
Improve publisher login and add publisher logout

### PR DESCRIPTION
**Pull request by ruwiniwj**
This PR contains the following;
1. Add redirect upon login in to publisher (https://github.com/wso2/product-apim/issues/4960)
2. Add publisher logout (https://github.com/wso2/product-apim/issues/5106)
3. Introduce a util class 'jagg.jag' to get the server port rather than using global  object 'application' 
4. Remove unnecessary logs in jag files
5. Add missing license headers

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    include("/services/constants.jag");
+
+    var utils = Packages.org.wso2.carbon.apimgt.impl.utils.APIUtil;
+    var serverPort = utils.getManagementTransportPort(MGT_TRANSPORT.split(":")[0]);
+
+    var getServerPort = function () {
+        return serverPort;
+    };
+
+%>
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag]: Add a new line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/logout/logout_callback.jag [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        'path': site.context + "/services/logout",
+        "secure": true,
+        "maxAge": 2
+    };
+    response.addCookie(cookie);
+
+    log.debug("redirecting to login");
+    response.sendRedirect(site.context + "/login");
+
+%>
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/logout/logout_callback.jag]: Add a new line

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/services/jagg.jag [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    include("/services/constants.jag");
+
+    var utils = Packages.org.wso2.carbon.apimgt.impl.utils.APIUtil;
+    var serverPort = utils.getManagementTransportPort(MGT_TRANSPORT.split(":")[0]);
+
+    var getServerPort = function () {
+        return serverPort;
+    };
+
+%>
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/services/jagg.jag]: Add a new line

Comment (ruwiniwj) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/services/jagg.jag]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/logout/logout_callback.jag (hunk 2 above)
Comment (ruwiniwj) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/logout/logout_callback.jag]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag (hunk 1 above)
Comment (ruwiniwj) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+    include("/services/constants.jag");
+
+    var utils = Packages.org.wso2.carbon.apimgt.impl.utils.APIUtil;
+    var serverPort = utils.getManagementTransportPort(MGT_TRANSPORT.split(":")[0]);
```
Comment (chamilaadhi) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag]: does this work when there is a url without a port? say LB address

Comment (ruwiniwj) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/services/jagg.jag]: The method `.getManagementTransportPort()` returns the port for the given transport type, value of `MGT_TRANSPORT` is set to "https://" in constants.jag.

**Pull request merged**